tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15746,6 +15782,8 @@ void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD,
     return;
   }
 
+  isActiveEntry = true;
+
   std::string profile = S.getLangOpts().HLSLProfile;
   const ShaderModel *SM = hlsl::ShaderModel::GetByName(profile.c_str());
   const llvm::StringRef fullName = ShaderModel::FullNameFromKind(SM->GetKind());
```

Comment:
Below here we have:
```cpp
  // don't add the attribute for an invalid profile, like library
  if (fullName.empty()) {
    return;
  }
```

However, I believe this should be an assert or llvm_unreachable, since we shouldn't have called this function in that case.